### DISCUSSION TITLE
Shared ui components

### DISCUSSION THREAD

**Discussion by lnwara** [keycloak/keycloak#24461] [Q&A]
### Description

I was trying to configure a simple custom theme and I had face an issue of creating shared components and shared scripts that could be accessed from any ui module (login, account, etc..). Unfortunately, after a while of search I don't get any valid solution.

for example, I have the following scenario:

I need to have shared header.ftl, footer.ftl and shared script in Common folder, these files should be accessible through any module inside my theme folder, but I get Backing out from the root directory is not allowed.

Whenever the file (header.ftl) is inside login folder it accessible but if I place it outside Login in Common folder and access it as

<#include "../Common/header.ftl">

I get this error logs

Error executing FreeMarker template: freemarker.core._MiscTemplateException: Template inclusion failed (for parameter value "../common/header.ftl"): 2023-10-30 12:45:27 Template not found for name "../common/header.ftl". 2023-10-30 12:45:27 Reason given: Backing out from the root directory is not allowed.

### Motivation

I'm not sure if there is any kind of solution for this issue, but it would be more mainatainable if we could implement a shared components.

**Comment by jonkoops** [comment]
Converted this to a discussion as it's not really a clear bug or feature request.

**Comment by ssilvert** [comment]
Support for this is on our road map.

Basically, we see three types of common resources depending on access scope.

- Available from any theme and any realm (global)
- Available from the realm
- Available from the theme
